### PR TITLE
Feature/Teams command

### PR DESCRIPTION
**Pull request by erbenjak**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +34,6 @@
   },
   "publishConfig": {
     "access": "public",
-    "provenance": true
+    "provenance": false
```
Comment (jkoenig134) [inline package.json:37]: re-add provenance

File: package.json
Comment (jkoenig134) [inline package.json:1]: missing version bump (make version `0.0.2` and run `npm i`)

File: src/teams/teams-messaging.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        for (var urlToCheck in urls){
+            try {
+                new URL(urlToCheck);
+            } catch (err) {
+                return false;
+            }
+        }
+        return false;
+    }
+}
```
Comment (jkoenig134) [inline src/teams/teams-messaging.ts:67]: missing newline at the end of the file

File: src/index.ts [hunk 3]
```diff
@@ -1,9 +1,26 @@
 #!/usr/bin/env node
 
 import yargs from "yargs"
+import { TeamsMessaging } from "./teams/teams-messaging";
+
+
+const args = require("yargs").usage("jscm command")
+    .command( 'teams','This command is used to send a teams message via a passed webhook',
```
Comment (jkoenig134) [inline src/index.ts:8]: formatting is **very** off here. have you installed https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode ?

File: src/teams/teams-messaging.ts
Comment (jkoenig134) [inline src/teams/teams-messaging.ts:1]: this should be `src/commands/TeamsMessaging.ts`

Comment (jkoenig134) [reply src/teams/teams-messaging.ts:1]: you missed the folder rename

File: src/commands/TeamsMessaging.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import axios from "axios"
+
+export class TeamsMessaging {
+  async main(webhook: string, platformIdent: string, artifactLink: string, isSuccessful: boolean) {
```
Comment (jkoenig134) [inline src/commands/TeamsMessaging.ts:4]: what does `platformIdent` mean?

File: src/commands/TeamsMessaging.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    if (!this.isSanityCheckUrlsOk([webhook, artifactLink])) {
+      console.log("ERROR: A given Url is not a valid url.")
```
Comment (jkoenig134) [inline src/commands/TeamsMessaging.ts:6]:
```suggestion
      console.log("ERROR: A given Url is not valid.")
```

File: src/commands/TeamsMessaging.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      return
+    }
+
+    var statusIdentifier = "Successful"
+    if (!isSuccessful) {
+      statusIdentifier = "Failed"
+    }
```
Comment (jkoenig134) [inline src/commands/TeamsMessaging.ts:13]:
```suggestion
    const statusIdentifier = isSuccessful ? "Successful" : "Failed"
```

File: src/commands/TeamsMessaging.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+      }
+    }
+    return true
+  }
+
+  private getTextTeamsMsg(isSuccessful: boolean) {
+    if (isSuccessful) {
+      return "The newly released version did build and is now available as an artifact."
+    } else {
+      return "The newly released version did NOT build and the logs now available as an artifact."
```
Comment (jkoenig134) [inline src/commands/TeamsMessaging.ts:70]: maybe just make this inline with a ternary operator => `isSuccessful ? "bla" : "blub"`

Comment (jkoenig134) [reply src/commands/TeamsMessaging.ts:70]: also you can remove the function then.

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

File: src/commands/TeamsMessaging.ts (hunk 4 above)
Comment (erbenjak) [reply src/commands/TeamsMessaging.ts]: platform-identifier -> renamed to platform

File: src/commands/TeamsMessaging.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+          type: "string"
+        },
+        artifactUrl: {
+          description: "download link for the generated artifact (logs or build)",
+          required: true,
+          type: "string"
+        },
+        wasBuildSuccessful: { description: "status of the finished build", required: true, type: "boolean" }
+      })
+      .choices("platform", ["ios", "android"])
```
Comment (jkoenig134) [inline src/commands/TeamsMessaging.ts:57]: move choices up to the platform object

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**